### PR TITLE
Return last fedimint error if node manager fails

### PR DESCRIPTION
**Pull request by TonyGiorgio**
Eventually I think we can stop making assumptions about the node manager being there at all, but that refactor for another day.

### CODE REVIEW

File: mutiny-core/src/lib.rs [hunk 1]
```diff
@@ -776,6 +779,22 @@ impl<S: MutinyStorage> MutinyWallet<S> {
...
+                    // InsufficientBalance for the node manager might just mean there is
+                    // no lightning node balance at all, something that might be common for
+                    // federation only users. Attempt to take a federation error if that
+                    // was configured and InsufficientBalance is returned.
+                    MutinyError::InsufficientBalance => {
+                        if let Some(federation_error) = last_federation_error {
+                            federation_error
+                        } else {
+                            e
+                        }
```
Comment (benthecarman) [inline mutiny-core/src/lib.rs:793]: this can be simplified as `last_federation_error.unwrap_or(e)`

File: mutiny-core/src/lib.rs [hunk 2]
```diff
@@ -776,6 +779,22 @@ impl<S: MutinyStorage> MutinyWallet<S> {
         self.node_manager
             .pay_invoice(None, inv, amt_sats, labels)
```
Comment (benthecarman) [inline mutiny-core/src/lib.rs:780]: should we just check the node manager balance first and see if we could even pay it?

Comment (TonyGiorgio) [reply mutiny-core/src/lib.rs:780]: seems like an unnecessary call that'll happen anyways and produce the same exact error

**Review by benthecarman — APPROVED** [review]

File: mutiny-core/src/lib.rs (hunk 2 above)
Comment (TonyGiorgio) [reply mutiny-core/src/lib.rs]: Decided it was an okay idea to go ahead and do. It'll do the call mostly twice but whatever.

**Review by benthecarman — APPROVED** [review]

**Pull request merged**